src/backend/dx12/src/command.rs

```diff
@@ -1147,34 +1150,70 @@ impl com::RawCommandBuffer<Backend> for CommandBuffer {
         T: IntoIterator,
         T::Item: Borrow<com::AttachmentClear>,
         U: IntoIterator,
-        U::Item: Borrow<pso::Rect>,
+        U::Item: Borrow<pso::ClearRect>,
     {
         assert!(self.pass_cache.is_some(), "`clear_attachments` can only be called inside a renderpass");
-        let rects: SmallVec<[d3d12::D3D12_RECT; 16]> = rects.into_iter().map(|rect| get_rect(rect.borrow())).collect();
+
+        let clear_rects: SmallVec<[pso::ClearRect; 16]> = rects.into_iter().map(|rect| *rect.borrow()).collect();
+
         for clear in clears {
             let clear = clear.borrow();
             match *clear {
                 com::AttachmentClear::Color(index, cv) => {
-                    let rtv = {
+                    let attachment = {
                         let pass_cache = self.pass_cache.as_ref().unwrap();
                         let rtv_id = pass_cache
                             .render_pass
                             .subpasses[self.cur_subpass]
                             .color_attachments[index]
                             .0;
 
-                        pass_cache
-                            .framebuffer
-                            .attachments[rtv_id]
-                            .handle_rtv
-                            .unwrap()
+                        let attachment =
+                            pass_cache
+                                .framebuffer
+                                .attachments[rtv_id];
+
+                        attachment
                     };
 
-                    self.clear_render_target_view(
-                        rtv,
-                        cv.into(),
-                        &rects,
-                    );
+                    let mut rtv_pool = n::DescriptorCpuPool {
```

Comment:
where is it getting freed? how do we ensure it's alive for the lifetime of the command buffer?
(@msiglreith correct me if this is not required here)

Reply:
> how do we ensure it's alive for the lifetime of the command buffer

AFAIK cpu descriptors don't need to live for the whole command buffer lifetime, just for encoding of the commands but don't quote me on this.

Reply:
Found this on msdn, but the current code seems to work for me at least
>  A handle itself can be copied freely or passed into APIs that operate on/use descriptors. There is no ref counting, so the application must ensure that it does not use a handle after the underlying descriptor heap has been deleted. 

https://msdn.microsoft.com/en-us/library/windows/desktop/dn859359(v=vs.85).aspx#Descriptor_Handles

Reply:
So it's a matter of what "use the handle" means exactly. Does "use" last for the execution of the command buffer on GPU? Or does it end when we issue the API calls?

Reply:
https://msdn.microsoft.com/en-us/library/windows/desktop/dn899199(v=vs.85).aspx

> After recording into the command list (with a call such as OMSetRenderTargets, for example) the memory used to hold the descriptors for this call is immediately available for re-use after the call.

Reply:
`self.rtv_pools` as they will/can be removed. renderdoc nightly works without storing those.
`temporary_gpu_heaps` are required to be linked with the lifetime of the command buffer.

Reply:
sorry, I incorrectly recalled which one here, thanks for correction!